### ISSUE TITLE
Diffing_oM: review the concept of Stream and Revision

### ISSUE THREAD

**Issue by alelom** [BHoM/BHoM#590]
In light of the definitions used also in the [REST API specification](https://github.com/aecdeltas/aec-deltas-spec/wiki/REST-API), the Stream class needs to be reviewed.

The Stream class should be renamed as `StreamRevision` as this is the only thing that can be produced in the UI.

- A revision is a "photograph" of a Stream. It represents all the Stream content at a given time. This can be represented through a type in the oM.

- A Stream is an abstract concept that represents a collection of objects that went through different revisions. I am not sure it make sense for us to have an oM type for it.

**Comment by al-fisher** [comment]
Agreed - as discussed - the `Stream` class as currently implemented is now closer to our concept of `Revision`.
I do think a (new) `Stream` class will actually be very useful though - on creation will simply be an empty container for Revisions - but can have its own engine methods to provide efficiency in creating a new revision - when "pushing" objects onto the stream etc.
